File: 14_longest_common_prefix/main.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prev_pref = strs[0]
        
        for i in range(1, len(strs)):
          word = strs[i]
          
          j = 0
          while j < len(prev_pref) and j < len(word) and prev_pref[j] == word[j]:
            j += 1
          prev_pref = prev_pref[:j]
        if prev_pref:
          return prev_pref
        else:
          return ""
              
          
example = Solution()
print(example.longestCommonPrefix(["dog","racecar","car"]))

File: 14_longest_common_prefix/main.go
package main

import "fmt"

func longestCommonPrefix(strs []string) string {
	current_pref := strs[0]

	for i := 1; i < len(strs); i++ {
		word := strs[i]

		j := 0
		for j < len(word) && j < len(current_pref) && word[j] == current_pref[j] {
			j++
		}
		current_pref = current_pref[0:j]

	}
	return current_pref

}

func main() {
	fmt.Printf("%s\n", longestCommonPrefix([]string{"ab", "a"}))
}
